lib/init.ts

```diff
@@ -12,21 +19,34 @@ export function init(options: InitParams) {
     throw new Error(
       "Init function should be called with an object argument containing your apiKey and applicationID"
     );
-  } else if (!options.apiKey || typeof options.apiKey !== "string") {
+  }
+  if (isUndefined(options.apiKey) || !isString(options.apiKey)) {
     throw new Error(
       "apiKey is missing, please provide it so we can authenticate the application"
     );
-  } else if (
-    !options.applicationID ||
-    typeof options.applicationID !== "string"
-  ) {
+  }
+  if (isUndefined(options.applicationID) || !isString(options.applicationID)) {
     throw new Error(
       "applicationID is missing, please provide it, so we can properly attribute data to your application"
     );
   }
+  if (
+    !isUndefined(options.region) &&
+    SUPPORTED_REGIONS.indexOf(options.region) === -1
+  ) {
+    throw new Error(
+      `optional region is incorrect, please provide either one of: ${SUPPORTED_REGIONS.join(
+        ", "
+      )}.`
+    );
+  }
 
   this._apiKey = options.apiKey;
   this._applicationID = options.applicationID;
+  this._region = options.region;
+  this._endpointOrigin = options.region
+    ? `https://insights.${options.region}.algolia.io`
+    : "https://insights.algolia.io";
```

Comment:
It means that we can't provide a specific endpoint (e.g. testing) unless we override the `_endpointOrigin`. We could use `ENV` variables to define the endpoint rather than having it hard coded in the source code.

Reply:
yes, I didn't want to expose it because it's not part of the spec.
Yes it's a problem for testing, but there is a workaround so I didn't mind. 

If we expose something to set an endpoint, I suspect we'd have to document how it overrides regions. I thought it was simpler this way.

Reply:
I was not talking about exposing an API but more a replacement at build step something like that.